.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          # Just some default versions to run code quality tools with.
+          # These can be upgraded as and when but probably won't need to change much.
+          otp-version: "24.0.6"
+          elixir-version: "1.12.3"
+      - uses: actions/cache@v2
+        with:
+          path: |
+            deps
+            _build
+            priv/plts
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - run: mix deps.get
+      - run: mix credo --strict
+        continue-on-error: true
+      - run: mix format --check-formatted
+        continue-on-error: true
+      - run: mix dialyzer
+  test:
+    strategy:
+      matrix:
+        os: 
+          - ubuntu-latest
+        otp-version:
+          - "24.0.6"
+        elixir-version: 
+          - "1.12.3"
+        mongo-version:
+          - maj: 4
+            min: 4
+            patch: 8
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          otp-version: ${{ matrix.otp-version }}
+          elixir-version: ${{ matrix.elixir-version }}
+      - uses: actions/cache@v2
+        with:
+          path: |
+            deps
+            _build
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - name: Install Mongo
+        run: |
+          # See https://docs.mongodb.com/manual/tutorial/install-mongodb-on-ubuntu/
+
+          # Import MongoDB public key
+          wget -qO - https://www.mongodb.org/static/pgp/server-${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }}.asc | sudo apt-key add -
+
+          # Create a list file for MongoDB
+          echo "deb [ arch=amd64,arm64 ] https://repo.mongodb.org/apt/ubuntu focal/mongodb-org/${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }} multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }}.list
+
+          # Update package db
+          sudo apt-get update
+
+          # Finally install mongodb
+          sudo apt-get install -y \
+            --allow-downgrades \
+            mongodb-org=${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }}.${{ matrix.mongo-version.patch }} \
+            mongodb-org-server=${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }}.${{ matrix.mongo-version.patch }} \
+            mongodb-org-shell=${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }}.${{ matrix.mongo-version.patch }} \
+            mongodb-org-mongos=${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }}.${{ matrix.mongo-version.patch }} \
+            mongodb-org-tools=${{ matrix.mongo-version.maj }}.${{ matrix.mongo-version.min }}.${{ matrix.mongo-version.patch }}
```

Comment:
Maybe instead of this, you can look at what [one of the official mongo drivers](https://github.com/mongodb/mongo-ruby-driver/blob/master/.github/workflows/test.yml) uses to start mongo.

Reply:
Oh excellent find.  Agreed.